hyperbahn/client.go

```diff
@@ -102,6 +102,9 @@ func parseConfig(config *Configuration) error {
 		if err := decoder.Decode(&config.InitialNodes); err != nil {
 			return err
 		}
+		if err = f.Close(); err != nil {
```

Comment:
Can you move this to a defer after the file is opened. That way, even if decoding fails, the file will still be closed.
```go
f, err := os.Open(config.InitialNodesFile)
if err != nil {
  return err
}
defer f.Close()
```